### PR TITLE
Separate SMART Response

### PR DESCRIPTION
**Pull request by emichaud998**
Pull requests into this repository require the following checks to be completed by the submitter and reviewers.
Changes in this PR:
This PR separates the SMART Response code from the Capability Statement code in the capability parser factory while also adding some testing for the SMART Response Equal function using SMART responses from the production DB.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [Lantern 300 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-300)
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [ ] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [ ] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

File: endpointmanager/pkg/endpointmanager/smartresponse.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+		if err != nil {
+			return false
+		}
+		resp2Copy, err = deleteFieldFromSmartResponse(resp2Copy, field)
+		if err != nil {
+			return false
+		}
+	}
+
+	j1, err := respCopy.GetJSON()
```
[the remark this answers is not in the document]
Comment (emichaud998) [reply endpointmanager/pkg/endpointmanager/smartresponse.go]: You are totally right! Fixed!

**Comment by bmath10** [comment]
I know Matt has already mentioned the issues he saw on the SMART Response page, and this might be related but I just wanted to mention it anyways: I just ran with the list of endpoints that Matt posted in the #lantern slack channel from the connectathon and i had two endpoints return valid SMART responses, but the two endpoints listed on the SMART Response page were two different URLs that had responded with null.
Screen Shot 2021-02-16 at 9 27 24 AM
Screen Shot 2021-02-16 at 9 26 25 AM

File: shinydashboard/lantern/modules/smartresponsemodule.R [hunk 2]
```diff
@@ -48,6 +48,85 @@ smartresponsemodule <- function(
...
 
+  selected_smart_count_total <- reactive ({
+    all <- endpoint_export_tbl
+    req(sel_fhir_version(), sel_vendor())
+    if (sel_fhir_version() != ui_special_values$ALL_FHIR_VERSIONS) {
+      all <- all %>% filter(fhir_version == sel_fhir_version())
+    }
+    if (sel_vendor() != ui_special_values$ALL_DEVELOPERS) {
+      all <- all %>% filter(vendor_name == sel_vendor())
+    }
```
Comment (bmath10) [inline shinydashboard/lantern/modules/smartresponsemodule.R:59]: It seems like you could possibly make this into a function that all of these other functions call so you don't have to write it out every time? something like:
```
get_filtered_data <- function(table_val) {
   all <- table_val
   req(sel_fhir_version(), sel_vendor())
   ....
   all
}
```
at least I think that would be possible in R?

**Comment by bmath10** [comment]
I'm currently getting this error when running the shiny dashboard:
```
Warning: Error in get_well_known_endpoint_counts: could not find function "get_well_known_endpoint_counts"
shinydashboard_1       |   91: force
shinydashboard_1       |   90: app_data$well_known_endpoint_counts
shinydashboard_1       |   89: <reactive>
shinydashboard_1       |   73: database_fetcher
shinydashboard_1       |   72: observeEventHandler [/srv/shiny-server/server.R#21]
shinydashboard_1       |    1: runApp
shinydashboard_1       |
```
I'm guessing because `get_well_known_endpoint_counts` is still referenced at the bottom of `endpoints.R` even though it's no longer a function?

**Comment by emichaud998** [comment]
> I'm currently getting this error when running the shiny dashboard:
> 
> ```
> Warning: Error in get_well_known_endpoint_counts: could not find function "get_well_known_endpoint_counts"
> shinydashboard_1       |   91: force
> shinydashboard_1       |   90: app_data$well_known_endpoint_counts
> shinydashboard_1       |   89: <reactive>
> shinydashboard_1       |   73: database_fetcher
> shinydashboard_1       |   72: observeEventHandler [/srv/shiny-server/server.R#21]
> shinydashboard_1       |    1: runApp
> shinydashboard_1       |
> ```
> 
> I'm guessing because `get_well_known_endpoint_counts` is still referenced at the bottom of `endpoints.R` even though it's no longer a function?

Interesting, I did not see this when I was running it but maybe I missed it, I will take a look now!

**Review by bmath10 — APPROVED** [review]

**Pull request merged**